src/dawn/engine_dawn.cc

```diff
@@ -791,27 +796,31 @@ Result DawnPipelineHelper::CreateRenderPipelineDescriptor(
   }
   // Fill the default values for vertexInput.
   // assuming #inputs = #attributes
-  // if isDrawRectCommand == true ignore index/vertex buffers that are are
-  // attached to the pipeline and attach one buffer with format of Float4  and
-  // 4*sizeof(float) stride (format and stride of the buffer that we create in
-  // DoDrawRect)
   int num_inputs = 0;
   for (uint32_t i = 0; i < kMaxVertexInputs; ++i) {
-    if (isDrawRectCommand && i == 0) {
-      vertexInput.inputSlot = i;
-      vertexInput.stride = 4 * sizeof(float);
-      vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
-      num_inputs = 1;
-    } else if (i < render_pipeline.pipeline->GetVertexBuffers().size()) {
-      vertexInput.inputSlot = i;
-      vertexInput.stride = render_pipeline.pipeline->GetVertexBuffers()[i]
-                               .buffer->GetTexelStride();
-      vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
-      num_inputs = render_pipeline.pipeline->GetVertexBuffers().size();
+    if (ignore_vertex_and_Index_buffers) {
+      if (i == 0) {
+        vertexInput.inputSlot = 0;
+        vertexInput.stride = 4 * sizeof(float);
+        vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+        num_inputs = 1;
+      } else {
+        vertexInput.inputSlot = 0;
+        vertexInput.stride = 0;
+        vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+      }
     } else {
-      vertexInput.inputSlot = 0;
-      vertexInput.stride = 0;
-      vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+      if (i < render_pipeline.pipeline->GetVertexBuffers().size()) {
+        vertexInput.inputSlot = i;
+        vertexInput.stride = render_pipeline.pipeline->GetVertexBuffers()[i]
+                                 .buffer->GetTexelStride();
+        vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+        num_inputs = render_pipeline.pipeline->GetVertexBuffers().size();
```

Comment:
This is a redundant lookup, but harmless this way.  It's very clear what's going on, and that's good.